src/delay.rs

```diff
@@ -1,91 +1,29 @@
-//! Delays
-use core::cmp;
-use cortex_m::peripheral::SYST;
-use hal::blocking::delay::{DelayMs, DelayUs};
-
-use crate::prelude::*;
 use crate::rcc::Clocks;
-use crate::time::{Hertz, MicroSecond};
-
-/// System timer (SysTick) as a delay provider
-pub struct Delay {
-    clk: Hertz,
-    syst: SYST,
-}
-
-impl Delay {
-    /// Configures the system timer (SysTick) as a delay provider
-    pub fn new(syst: SYST, clocks: &Clocks) -> Self {
-        Delay {
-            syst,
-            clk: clocks.ahb_clk / 8,
-        }
-    }
-
-    pub fn delay<T>(&mut self, delay: T)
-    where
-        T: Into<MicroSecond>,
-    {
-        let mut cycles = delay.into().cycles(self.clk);
-        while cycles > 0 {
-            let reload = cmp::min(cycles, 0x00ff_ffff);
-            cycles -= reload;
-            self.syst.set_reload(reload);
-            self.syst.clear_current();
-            self.syst.enable_counter();
-            while !self.syst.has_wrapped() {}
-            self.syst.disable_counter();
-        }
-    }
-
-    /// Releases the system timer (SysTick) resource
-    pub fn release(self) -> SYST {
-        self.syst
-    }
-}
-
-impl DelayUs<u32> for Delay {
-    fn delay_us(&mut self, us: u32) {
-        self.delay(us.us())
-    }
-}
-
-impl DelayUs<u16> for Delay {
-    fn delay_us(&mut self, us: u16) {
-        self.delay_us(us as u32)
-    }
-}
-
-impl DelayUs<u8> for Delay {
-    fn delay_us(&mut self, us: u8) {
-        self.delay_us(us as u32)
-    }
-}
+use crate::time::MicroSecond;
+pub use cortex_m::delay::*;
+use cortex_m::{peripheral::SYST, prelude::_embedded_hal_blocking_delay_DelayUs};
 
-impl DelayMs<u32> for Delay {
-    fn delay_ms(&mut self, ms: u32) {
-        self.delay_us(ms.saturating_mul(1_000));
-    }
-}
-
-impl DelayMs<u16> for Delay {
-    fn delay_ms(&mut self, ms: u16) {
-        self.delay_ms(ms as u32);
-    }
+pub trait SYSTDelayExt {
```

Comment:
Renamed this because the `DelayExt` would fit more for extensions of `Delay` since this crate isn't yet widely used and is `v0.0.0`, changes like this could be acceptable.